### 7.2. Calibration Results

This is organized in the following way

- [MATLAB calibration routine](#cal)

- [Zeta normalization analysis](#zeta)

---
### <a name="cal"></a> Calibration Routine

The code below access the commandline/terminal and then directly executes the MATLAB code. It assumes that (i) you have a working installation of MATLAB and (ii) that it is in your path. Also verify that the directory change implemented below is correct.

The resulting parameters are outputed to [``/parameters/calibration_params.csv``](/parameters/calibration_params.csv)

The underlying MATLAB code is completly described (with links to the appropriate ``.m`` files) in the [readme file in the calibraiton folder](/src/calibration/README.md).

In [83]:
matlab_path = '''matlab'''
# This assumes that Matlab is in your path. If not then direct it to
# program files / matlab / version/ bin/ matlab.exe
# needs to be in triple quotes


matlab_cmd = '''"cd('src\calibration');calibrate_wrap;"'''
# then the way this works is you have to give the matlab command to change the working directory
# this is the cd part; then call the function which in this case is calibrate_wrap
# needs to be in triple quotes 

In [84]:
!{matlab_path} -batch {matlab_cmd}

# Then the ! command passes the stuff to the command prompt
# or terminal. the {} remove the triple quotes
# -batch runs this stuff in the background, i.e. no matlab
# terminal is opend, it will autormatically exit and close matlab
# then the next {} is function call to matlab

   14-Apr-2020

Calibration Targets...
Real Rate and Productivity Growth
    0.0283    0.0079

BEJK Exporter Moments: Fraction of Exporters, Relative Size
    0.0330    4.8000

Home Trade Share
    0.8937

Entry Moment
    0.0200

Firm Moments (LHS Table 3)
    0.5010    0.2700    0.1360    0.0940
    0.1610    0.2850    0.2900    0.2600

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
Optimization terminated:
 the current x satisfies the termination criteria using OPTIONS.TolX of 1.000000e-04 
 and F(X) satisfies the convergence criteria using OPTIONS.TolFun of 1.000000e-04 

Elapsed time is 39.260284 seconds.
    0.0037

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Calibration Results
Parameter Values
d, theta, kappa, 1/chi, mu, upsilon, sigma, delta, rho
    3.0225    4.9890    0.1042    7.8835   -0.0311    0.0483    3.1669    0.0200    0.0203

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Mom

In [85]:
import pandas as pd

col_list = [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15]
name_list = ["d_0", "θ", "κ", "χ", "μ", "υ", "σ", "δ", "ρ", "ζ", "N"  ,"η", "Theta", "γ", "d_T"]

name_dict = dict(zip(col_list,name_list))

In [86]:
def fix_headings(path, name_dict):
    
    df = pd.read_csv(path, header = None)

    df = df.rename(axis = 1, mapper = name_dict)

    df.to_csv(path)
    
    return df

In [87]:
raw_path = "parameters\calibration_params.csv"

df = fix_headings(raw_path,name_dict)

In [88]:
df.T

0
d_0     3.022493
θ       4.988977
κ       0.104196
χ       0.126847
μ      -0.031065
υ       0.048301
σ       3.166924
δ       0.020000
ρ       0.020338
ζ       1.000000
N      10.000000
η       0.000000
Theta   1.000000
γ       1.000000
d_T     2.820244

Again, output is at [``/parameters/calibration_params.csv``](/parameters/calibration_params.csv) which contains the parameter values in the with headers for each column describing the associated parameter. 

#### Brief Commentary (same as in paper)

In terms of fit, the aggregate moments (real rate and productivity growth), the exporter facts, and the employment share of entrants match
the data exactly. The model’s fit for the eight firm dynamic moments are the final outputs  While not perfect, the model
replicates the dynamics of firms well. The correlation between the model and the data is extremely high at 0.98.

---
### <a name="zeta"></a> Zeta Normalization Analysis

In Section 7.2, we discuss our normalization of zeta and how our results do not depend on the particular normalization. The discussion is below...

As stated earlier, we set ζ = 1 as a normalization. Several comments are in order regarding this normalization. First, calibrating the model with a different normalization results in identical parameter values for all parameters except κ. In the theoretical analysis, equations (33) and (35) show that ζ only appears in the form ζ/κ. Thus, a different choice of ζ results in a different κ to match the g and λii moments.
Different normalizations do lead to a different amount of variety as equation (48) shows. However, as equation (53) shows, any combination of ζ and κ results in the same welfare gains from trade as long as λii and all other parameter values are equal. It is in this sense that ζ = 1 is a normalization. Numerical analysis of the quantitative model confirms these observations both for steady-state comparative statics
and for the analysis of transition dynamics

#### Calibration under alternative  ζ

In [89]:
matlab_cmd = '''"cd('src\calibration');calibrate_wrap_zeta;"'''
# then the way this works is you have to give the matlab command to change the working directory
# this is the cd part; then call the function which in this case is calibrate_wrap_zeta
# needs to be in triple quotes 

In [90]:
!{matlab_path} -batch {matlab_cmd}

   14-Apr-2020

Calibration with different zetas
Zeta Value
    0.7500

 
Optimization terminated:
 the current x satisfies the termination criteria using OPTIONS.TolX of 1.000000e-04 
 and F(X) satisfies the convergence criteria using OPTIONS.TolFun of 1.000000e-04 

Elapsed time is 38.802036 seconds.
Parameter Values
d, theta, kappa, 1/chi, mu, upsilon, sigma, delta, rho, zeta
    3.0225    4.9890    0.0781    7.8835   -0.0311    0.0483    3.1669    0.0200    0.0203    0.7500

zeta/kappa
    9.5973

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Moments: Model Alt Zetas
Calibration Targets...
Real Rate and Productivity Growth
    0.0283    0.0079

BEJK Exporter Moments: Fraction of Exporters, Relative Size
    0.0330    4.7999

Home Trade Share
    0.8937

Entry Moment
    0.0200

Firm Moments
    0.2587    0.2883    0.0861    0.3148
    0.2823    0.1706    0.1418    0.4573

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Zeta 

In [91]:
raw_path = "parameters\calibration_zeta_0.75.csv "

fix_headings(raw_path, name_dict)

raw_path = "parameters\calibration_zeta_1.25.csv "

fix_headings(raw_path, name_dict)

d_0         θ         κ         χ         μ         υ         σ     δ  \
0  3.022493  4.988977  0.130245  0.126847 -0.031065  0.048301  3.166924  0.02   

          ρ     ζ   N  η  Theta  γ       d_T  
0  0.020338  1.25  10  0      1  1  2.820244

Note the calibrated parameter values are all in [``/parameters/calibration_zeta_X.XX.csv``](/parameters) and each ``.csv`` has the final endings of XX which coresponds with the zeta value assumed. The csv file which contains the parameter values in the following order: d, theta, kappa, 1/chi, mu, upsilon, sigma, delta, rho, zeta.

#### Brief Commentary

Per our claim in the paper: The calibration results lead to (i) the same exact fit (ii) all parameter values are the same except for kappa and (iii) the ratio of zeta/kappa is always the same (9.5973)

The next set of code confirms that this does not matter for counterfactualls or welfare.

#### Alternative zetas: counterfactuals and welfare.

**TO DO** 

1. Clean up julia notebook so it prints key issue

2. Set up julia notebook so it inputs zeta calibrations from and be clearer about warnings if running through this notebook.

**Warnings**

1. If you want you can just run the notebook [``ZetaNormalization.ipynb``](ZetaNormalization.ipynb)

2. An alternative is to just execute the code below (TO DO IS HAVE THE NBCONVERT CODE)

3. If this is hte first time running the julia code, it may take awhile to run.

4. Path needs to be correct

In [92]:
!jupyter nbconvert --to script ZetaNormalization.ipynb

[NbConvertApp] Converting notebook ZetaNormalization.ipynb to script
[NbConvertApp] Writing 4903 bytes to ZetaNormalization.jl


In [93]:
julia_path = '''C:\\Users\\mwaugh\\AppData\\Local\\Programs\\Julia\\Julia-1.4.0\\bin\\julia.exe'''
julia_command = '''ZetaNormalization.jl'''

In [94]:
!{julia_path} {julia_command}

g = 0.007913401963163568
Ω = 0.6805904609927731
z_hat = 1.9813268967445283
λ_ii = 0.8937087282949208
c = 0.870683493910354
g = 0.010250822794505147
Ω = 0.6459867736680399
z_hat = 1.8456817342356466
λ_ii = 0.8555713586729671
c = 0.8629419672035079
solving without cache.
480.242390 seconds (1.79 G allocations: 694.061 GiB, 15.40% gc time)
ce_1_transition = 100 * consumption_equivalent(solved.U[1], stationary_0.U_bar, overall_parameters) = 10.802138416380114
g = 0.007913401963163787
Ω = 0.9074539479905925
z_hat = 1.9813268967445021
λ_ii = 0.8937087282949173
c = 0.9943001206514334
g = 0.010250822798546106
Ω = 0.8613156829790773
z_hat = 1.845681734222669
λ_ii = 0.855571358670515
c = 0.985459476381832
solving without cache.
460.267494 seconds (1.81 G allocations: 698.087 GiB, 15.15% gc time)
ce_075_transition = 100 * consumption_equivalent(solved.U[1], stationary_0.U_bar, overall_parameters) = 10.964426594695208
g = 0.00791340196314654
Ω = 0.5444723687853111
z_hat = 1.98132689674431